packages/react-instantsearch-hooks-dom/src/ui/__tests__/Menu.test.tsx

```diff
@@ -0,0 +1,263 @@
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { Menu } from '../Menu';
+
+import type { MenuProps } from '../Menu';
+
+describe('Menu', () => {
+  function createProps(props: Partial<MenuProps>): MenuProps {
+    let isShowingMore = false;
+    return {
+      items: [
+        {
+          value: 'Insignia™',
+          label: 'Insignia™',
+          count: 746,
+          isRefined: true,
+        },
+        {
+          value: 'Samsung',
+          label: 'Samsung',
+          count: 633,
+          isRefined: false,
+        },
+      ],
+      onRefine: jest.fn(),
+      createURL: (value) => `#${value}`,
+      canToggleShowMore: true,
+      toggleShowMore: () => {
+        isShowingMore = !isShowingMore;
+      },
```

Comment:
Not sure this is necessary to reimplement the behavior of the function that the widget will pass. I'd make this a mock function without any implementation, then have a test that clicks the button and ensures it has been called. wdyt?

Reply:
It seems simpler to me to have a correct implementation than a mock, as the real rendering state is based on isShowingMore, so that would still need to be updated. This is closer to real-life usage :)

Reply:
If you want to test both (function called + updated state) then go ahead! It can still be in a mock though 🙂